client/src/containers/AuthContainer.test.tsx

```diff
@@ -91,119 +89,155 @@ describe("verifying valid input for signup page", () => {
+
+        const form = screen.getByTestId("form");
+        const mockSignUp = jest
+          .spyOn(ServerApi.prototype, "signUp")
+          .mockImplementation();
+        const passwordForm = screen.getByTestId("passwordForm");
+
+        fireEvent.blur(passwordForm);
+        fireEvent.submit(form);
+        /**
+         * Since fireEvent.submit is not async function we cannot use await
+         * Without an await, expect(mockSignUp) assertion can run before
+         * fireEvent.submit can complete execution causing the test to fail.
+         * Therefore, we await on dummy Promise to delay execution while
+         * fireEvent.submit completes.
+         */
+        await new Promise((r) => setTimeout(r, 1));
+        expect(mockSignUp).not.toBeCalled();
+      });
     });
 
-    it("should show email confirmation alert on success", async () => {
-      screen.getByTestId("SignUpTabButton").click();
-
-      const form = screen.getByTestId("form");
-      const mockSignUp = jest
-        .spyOn(ServerApi.prototype, "signUp")
-        .mockImplementation(() =>
-          Promise.resolve({ response: {} as AxiosResponse })
-        );
-      const mockAlert = jest
-        .spyOn(window, "alert")
-        .mockImplementation(() => {});
-
-      fireEvent.submit(form);
-      await new Promise((r) => setTimeout(r, 1));
-      expect(mockSignUp).toBeCalled();
-      expect(mockAlert).toBeCalledWith("Email confirmation sent!");
+    describe("On success", () => {
+      it("should call signUp if fields are valid", async () => {
+        screen.getByTestId("SignUpTabButton").click();
+
+        const fields = {
+          emailForm: "daniel.zingaro@mail.utoronto.ca",
+          userNameForm: "zingarod",
+          passwordForm: "password",
+          confirmPassForm: "password",
+          firstNameForm: "Daniel",
+          lastNameForm: "Zingaro",
+        };
+        Object.keys(fields).forEach((key) => {
+          fireEvent.change(
+            screen.getByTestId(key).querySelector("input") as HTMLInputElement,
+            { target: { value: fields[key as keyof typeof fields] } }
+          );
+        });
+
+        const form = screen.getByTestId("form");
+        const mockSignUp = jest
+          .spyOn(ServerApi.prototype, "signUp")
+          .mockImplementation(() =>
+            Promise.resolve({ response: {} as AxiosResponse })
+          );
+        const formData = {
+          email: "daniel.zingaro@mail.utoronto.ca",
+          userName: "zingarod",
+          password: "password",
+          firstName: "Daniel",
+          lastName: "Zingaro",
+        };
+
+        await act(async () => {
+          fireEvent.submit(form);
+          await new Promise((r) => setTimeout(r, 1));
+        });
+        expect(mockSignUp).toBeCalledWith(formData);
+
+      });
     });
   });
 });
 
-describe("Login component", () => {
+describe("SignIn component", () => {
   describe("handleSubmit function", () => {
```

Comment:
I would rename this to be something more descriptive of the process being tested, not some inner function name while we're here.